### PR TITLE
device-twin-service helper

### PR DESCRIPTION
**Pull request by danhellem**
# Checklist
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-node/blob/main/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem
device_twin samples are confusing and not accurate. They require a separate service to post patch updates for desired properties.

# Description of the solution
- added a new helper project that can be used as a backend service
- updated device_twin samples
- added readme's

### CODE REVIEW

**Comment by vishnureddy17** [comment]
nit: IMO it'd be better to use async/await rather than callbacks but it might not be worth going back and redoing it.

**Review by vishnureddy17 — APPROVED** [review]

File: device/samples/helpers/device-twin-service/twin-service.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const iothub = require('azure-iothub');
+const connectionString = process.env.IOTHUB_CONNECTION_STRING;
+const deviceId = 'devDevice';
+
+if (connectionString === undefined) {
+  console.log('IoT Hub connection string not set');
+  process.exit(-1);
```
Comment (vishnureddy17) [inline device/samples/helpers/device-twin-service/twin-service.js:9]: Can you explain what a negative exit code does? I haven't seen this before.

Comment (danhellem) [reply device/samples/helpers/device-twin-service/twin-service.js]: I don't think it does anything special :)

i changed it to zero

**Comment by danhellem** [comment]
> nit: IMO it'd be better to use async/await rather than callbacks but it might not be worth going back and redoing it.

If that is the better pattern then we should most def fix them. But let's do that in a separate PR

File: device/samples/helpers/device-twin-service/twin-service.js (hunk 1 above)
Comment (vishnureddy17) [reply device/samples/helpers/device-twin-service/twin-service.js]: A 0 exit code means success so I think it should be non-zero

File: device/samples/helpers/device-twin-service/package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "iot-hub",
```
Comment (BertKleewein) [inline device/samples/helpers/device-twin-service/package.json:2]: >"name": "iot-hub", [](http://example.com/codeflow?start=2&length=18)

You should give this a better name and add it to lerna.json and the inputs.js file in the internals repo.  lerna.json ensures that the dependencies get linked for local development, and input.js ensures that the dependency versions get updated.

**Review by BertKleewein — APPROVED** [review]
:shipit:

File: device/samples/helpers/device-twin-service/package.json (hunk 2 above)
Comment (danhellem) [reply device/samples/helpers/device-twin-service/package.json]: @BertKleewein I updated the name and added to lerna.json. I can't find an inputs.js file

**Pull request merged**